src/package/cppwinrt/nuget/Microsoft.Windows.CppWinRT.targets

```diff
@@ -130,77 +134,30 @@ Copyright (C) Microsoft Corporation. All rights reserved.
-            <CppWinRTDynamicProjectReferences Remove="@(CppWinRTDynamicProjectReferences)"/>
-            <CppWinRTDynamicProjectReferences Include="@(_ResolvedNativeProjectReferencePaths)"
-                Condition= "$([System.String]::Copy('%(Identity)').EndsWith('.dll',true,null))"/>
-        </ItemGroup>
-        <Message Text="CppWinRTDynamicProjectReferences: @(CppWinRTDynamicProjectReferences)" Importance="$(CppWinRTVerbosity)"/>
-    </Target>
-
-    <!--Get dynamic library project reference WinMD files (including managed assemblies) for IDL processing and appx manifest inclusion-->
-    <Target Name="GetCppWinRTDynamicWinMDReferences"
-            DependsOnTargets="GetCppWinRTDynamicProjectReferences;$(GetCppWinRTDynamicWinMDReferencesDependsOn)"
-            Returns="@(CppWinRTDynamicWinMDReferences)">
-        <!--To evaluate wildcard expansion of an item collection, must use intermediate property-->
-        <ItemGroup>
-            <_CppWinRTDynamicWinMDReferenceWildcard Remove="@(_CppWinRTDynamicWinMDReferenceWildcard)"/>
-            <_CppWinRTDynamicWinMDReferenceWildcard Include="@(CppWinRTDynamicProjectReferences->'%(RootDir)%(Directory)*.winmd')"/>
-            <_CppWinRTDynamicWinMDReferenceWildcard Condition="'%(_ResolvedNativeProjectReferencePaths.ProjectType)'!='StaticLibrary'"
-                Include="@(_ResolvedNativeProjectReferencePaths->WithMetadataValue('FileType','winmd')->'%(RootDir)%(Directory)*.winmd')"/>
-        </ItemGroup>
-        <PropertyGroup>
-            <_CppWinRTDynamicWinMDWildcard>@(_CppWinRTDynamicWinMDReferenceWildcard)</_CppWinRTDynamicWinMDWildcard>
-        </PropertyGroup>
-        <ItemGroup>
-            <_CppWinRTDynamicWinMDReferences Include="$(_CppWinRTDynamicWinMDWildcard)"/>
-            <_CppWinRTManagedWinMDReferences Include="@(_ResolvedNativeProjectReferencePaths->WithMetadataValue('FileType',''))" Condition= "$([System.String]::Copy('%(Identity)').EndsWith('.winmd',true,null))" />
-            <CppWinRTDynamicWinMDReferences Remove="@(CppWinRTDynamicWinMDReferences)"/>
-            <CppWinRTDynamicWinMDReferences Include="@(_CppWinRTDynamicWinMDReferences);@(_CppWinRTManagedWinMDReferences)">
-                <WinMDPath>%(Identity)</WinMDPath>
-            </CppWinRTDynamicWinMDReferences>
+            <!-- Get static library project references -->
+            <_CppWinRTStaticProjectReferences Remove="@(_CppWinRTStaticProjectReferences)"/>
+            <_CppWinRTStaticProjectReferences Include="@(_ResolvedProjectReferencePaths)"
+                Condition= "'%(_ResolvedProjectReferencePaths.ProjectType)'=='StaticLibrary' AND '%(_ResolvedProjectReferencePaths.WinMDFile)' == 'true'"/>
+            <CppWinRTStaticLibraryProjectWinMDReferences Remove="@(CppWinRTStaticLibraryProjectWinMDReferences)"/>
+            <CppWinRTStaticLibraryProjectWinMDReferences Include="@(_CppWinRTStaticProjectReferences)">
+                <WinMDPath>%(FullPath)</WinMDPath>
```

Comment:
isn't this going to create a path to lib file, not a winmd?  (see original _CppWinRTStaticWinMDReferenceWildcard)

Reply:
oh, that's the _ResolvedProjectReferencePaths.WinMDFile filter?

---
In reply to: [301731509](https://github.com/microsoft/xlang/pull/491#discussion_r301731509) [](ancestors = 301731509)

Reply:
That's not what I've seen, but that is just one data point. I can try this in another project to verify.

Reply:
This is now guaranteed with the CppWInRTGetResolvedWinMD target.